ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CoreVerifierSpec.kt

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compilance.verification.core
+
+import io.kotlintest.matchers.string.shouldContain
+import io.kotlintest.matchers.string.shouldNotContain
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.StringSpec
+import io.mockk.every
+import io.mockk.mockk
+import org.codice.compliance.Common.Companion.buildDom
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCoreRefMessage
+import org.codice.compliance.SAMLCore_3_2_1_d
+import org.codice.compliance.utils.TestCommon
+import org.codice.compliance.verification.core.CoreVerifier
+import java.time.Duration
+import java.time.Instant
+
+@Suppress("StringLiteralDuplication", "MagicNumber")
+class CoreVerifierSpec : StringSpec() {
+    init {
+        val now = Instant.now()
+        val later = now.plus(Duration.ofMinutes(30))
+
+        val response = { statusCode: String ->
```

Comment:
✏️ If you use methods instead of closures, you can provide default values for the parameters. A use case could be that instead of passing `"urn:samlconf:fail"` to most of these closures, you can just call the method with not arguments.